# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.cpp
# Commit: 7a4e3f5f674a
# Full Hash: 7a4e3f5f674ad675fa084bf1f30a79a15e8c7d2a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r 9da00c1364a5 -r 7a4e3f5f674a dom/fetch/InternalResponse.cpp
--- a/dom/fetch/InternalResponse.cpp	Wed Jan 18 00:50:19 2023 +0000
+++ b/dom/fetch/InternalResponse.cpp	Wed Jan 18 00:50:19 2023 +0000
@@ -186,6 +186,30 @@
   return result;
 }
 
+ParentToChildInternalResponse InternalResponse::ToParentToChildInternalResponse(
+    NotNull<mozilla::ipc::PBackgroundParent*> aBackgroundParent) {
+  ParentToChildInternalResponse result(GetMetadata(), Nothing(),
+                                       UNKNOWN_BODY_SIZE, Nothing());
+
+  nsCOMPtr<nsIInputStream> body;
+  int64_t bodySize;
+  GetUnfilteredBody(getter_AddRefs(body), &bodySize);
+
+  if (body) {
+    result.body() = Some(
+        ToParentToChildStream(WrapNotNull(body), bodySize, aBackgroundParent));
+    result.bodySize() = bodySize;
+  }
+
+  nsCOMPtr<nsIInputStream> alternativeBody = TakeAlternativeBody();
+  if (alternativeBody) {
+    result.alternativeBody() = Some(ToParentToChildStream(
+        WrapNotNull(alternativeBody), UNKNOWN_BODY_SIZE, aBackgroundParent));
+  }
+
+  return result;
+}
+
 SafeRefPtr<InternalResponse> InternalResponse::Clone(CloneType aCloneType) {
   SafeRefPtr<InternalResponse> clone = CreateIncompleteCopy();
   clone->mCloned = (mCloned = true);